### PR TITLE
Add DevMode panel (trigger: `CMD+'`)

### PR DESCRIPTION
**Pull request by zaiste**
I want to merge this change because it adds a DevMode panel with a GraphiQL playground that runs automatically authenticated queries.

**PR intended to be tested with API branch:** 

### Screenshots

Before | After
--- | ---
CleanShot 2023-03-13 at 15 30 58@2x | CleanShot 2023-03-13 at 15 33 43@2x

### Pull Request Checklist

1. [x] This code contains UI changes
2. [x] All visible strings are translated with proper context including data-formatting
3. [ ] Attributes `[data-test-id]` are added for new elements
4. [x] Changes are mentioned in the changelog
5. [x] The changes are tested in different browsers and in light/dark mode

### Acceptance Criteria

- [x] This feature adds a DevMode panel available under the `CMD+'` keystroke
- [x] In the DevMode panel, users can run GraphQL operations that are automatically authenticated
- [x] The playground in the DevMode panel has the same features as the one available from `<dashboard-rul>/graphql`
- [x] If users select `Open this order/product in GraphiQL` from the order/product details page, the query will be open in the playground of the DevMode panel

### Test environment config

API_URI=https://automation-dashboard.staging.saleor.cloud/graphql/
APPS_MARKETPLACE_API_URI=https://apps.staging.saleor.io/api/v2/saleor-apps
SALEOR_APPS_ENDPOINT=https://apps.saleor.io/api/saleor-apps

### Do you want to run more stable tests?

To run all tests, just select the stable checkbox. To speed up tests, increase the number of containers. Tests will be re-run only when the "run e2e" label is added.

1. [ ] stable
2. [ ] giftCard
3. [ ] category
4. [ ] collection
5. [ ] attribute
7. [ ] productType
8. [ ] shipping
9. [ ] customer
10. [ ] permissions
11. [ ] menuNavigation
12. [ ] pages
13. [ ] sales
14. [ ] vouchers
15. [ ] homePage
16. [ ] login
17. [ ] orders
18. [ ] products
19. [ ] app
20. [ ] plugins
21. [ ] translations
22. [ ] navigation
23. [ ] variants
24. [ ] payments

CONTAINERS=1

### CODE REVIEW

**Comment by michalina-graczyk** [comment]
Issues reported here https://github.com/saleor/saleor-dashboard/issues/3360 , please don't merge it without fix and re-testing.

File: src/components/DevModePanel/DevModePanel.tsx [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+        isDevModeVisible,
+        setDevModeVisibility,
+      }}
+    >
+      {children}
+    </DevModeContext.Provider>
+  );
+}
+
+interface Props {
```
Comment (poulch) [inline src/components/DevModePanel/DevModePanel.tsx:41]:
```suggestion
interface DevModePanelProps {
```

File: src/components/DevModePanel/DevModePanel.tsx [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+      open={isDevModeVisible}
+      style={{ zIndex: 5 }}
+      PaperProps={{ style: { height: "100%" } }}
+    >
+      <DialogHeader onClose={() => setDevModeVisibility(false)}>
+        Dev Mode
+      </DialogHeader>
+      {/* <div style={{ "margin":"20px" }}>
+        <Pill color="generic" label="customerCreate" onClick={() => {} }/>
+      </div> */}
```
Comment (poulch) [inline src/components/DevModePanel/DevModePanel.tsx:70]: suggestion: could be removed

**Comment by timuric** [comment]
font size is too large, should be 15px

**Comment by timuric** [comment]
Impossible to scroll documentation
image

File: src/components/DevModePanel/DevModePanel.tsx [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+export const DevModePanel: React.FC<DevModePanelProps> = ({
+  isDevModeVisible,
+  setDevModeVisibility,
+}) => {
+  const fetcher = createGraphiQLFetcher({
+    url: process.env.API_URI,
+    fetch: authorizedFetch,
+  });
+
+  const context = useDevModeContext();
```
Comment (poulch) [inline src/components/DevModePanel/DevModePanel.tsx:26]:
```suggestion
  const { devModeContent, variables } = useDevModeContext();
```

File: src/orders/components/OrderDetailsPage/OrderDetailsPage.tsx [hunk 4]
```diff
@@ -196,12 +196,20 @@ const OrderDetailsPage: React.FC<OrderDetailsPageProps> = props => {
...
+    context.setVariables(`{ "id": "${order?.id}" }`);
+    context.setDevModeVisibility(true);
+  };
+
+  // const _openPlaygroundURL2 = playgroundOpenHandler({
+  //   query: defaultGraphiQLQuery,
+  //   headers: "",
+  //   operationName: "",
+  //   variables: `{ "id": "${order?.id}" }`,
+  // });
```
Comment (poulch) [inline src/orders/components/OrderDetailsPage/OrderDetailsPage.tsx:212]: suggestion: Can be removed

File: src/products/components/ProductUpdatePage/ProductUpdatePage.tsx [hunk 5]
```diff
@@ -242,12 +242,20 @@ export const ProductUpdatePage: React.FC<ProductUpdatePageProps> = ({
...
+    context.setVariables(`{ "id": "${product?.id}" }`);
+    context.setDevModeVisibility(true);
+  };
+
+  // const openPlaygroundURL = playgroundOpenHandler({
+  //   query: defaultGraphiQLQuery,
+  //   headers: "",
+  //   operationName: "",
+  //   variables: `{ "id": "${product?.id}" }`,
+  // });
```
Comment (poulch) [inline src/products/components/ProductUpdatePage/ProductUpdatePage.tsx:258]: suggestion: can be removed

File: src/components/DevModePanel/hooks.ts [hunk 6]
```diff
@@ -0,0 +1,20 @@
+import { createContext, useContext } from "react";
+
+export const DevModeContext = createContext({
+  variables: "",
+  setVariables: (_value: string) => undefined,
+  isDevModeVisible: false,
+  setDevModeVisibility: (_value: boolean) => undefined,
+  devModeContent: "",
+  setDevModeContent: (_value: string) => undefined,
+});
```
Comment (poulch) [inline src/components/DevModePanel/hooks.ts:10]: suggestion: create an interface describing context values
```suggestion
export const DevModeContext = createContext<null | YourInterfcae>(null)
```

**Review by poulch — APPROVED** [review]

**Review by poulch — APPROVED** [review]
LGTM, btw I've seen that you ignored my previous suggestion xD

File: src/components/DevModePanel/DevModePanel.tsx [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+  return (
+    <Dialog
+      maxWidth="xl"
+      fullWidth
+      open={isDevModeVisible}
+      style={{ zIndex: 5 }}
+      PaperProps={{ style: { height: "100%" } }}
+    >
+      <DialogHeader onClose={() => setDevModeVisibility(false)}>
+        Dev Mode
```
Comment (krzysztofzuraw) [inline src/components/DevModePanel/DevModePanel.tsx:37]: **suggestion**: please add it to translations

File: src/components/GraphiQL/styles.ts [hunk 8]
```diff
@@ -57,7 +57,7 @@ export const useDashboardTheme = () => {
   const theme = useTheme();
 
   const rootStyle = {
-    "--font-size-body": theme.typography?.body2.fontSize,
+    "--font-size-body": theme.typography?.h3.fontSize,
```
Comment (krzysztofzuraw) [inline src/components/GraphiQL/styles.ts:60]: **question**: what about using styles from the new MacawUI?

File: src/components/GraphiQLPlain/index.ts [hunk 9]
```diff
@@ -0,0 +1,2 @@
+export { default } from "./GraphiQL";
```
Comment (krzysztofzuraw) [inline src/components/GraphiQLPlain/index.ts:1]: **suggestion**: what about using a named export instead of the default one?

Comment (zaiste) [reply src/components/GraphiQLPlain/index.ts:1]: @krzysztofzuraw I agree with you but here I made it how it's done for all other components at `src/components` in this codebase for consistency

**Review by poulch — APPROVED** [review]

**Review by poulch — DISMISSED** [review]

**Review by krzysztofzuraw — DISMISSED** [review]

**Review by krzysztofzuraw — APPROVED** [review]

**Review by poulch — APPROVED** [review]

**Pull request merged**